lib/routes/account.js

```diff
@@ -139,10 +139,32 @@ module.exports = (
           return db.deleteAccount(emailRecord)
         }
 
-        function ignoreUnknownAccountError (err) {
-          if (err.errno !== error.ERRNO.ACCOUNT_UNKNOWN) {
-            throw err
+        function checkAccountError (err) {
+          if (err.errno === error.ERRNO.ACCOUNT_UNKNOWN) {
+            // Even though the email is not on the account table, it could be on the emails table.
+            // Delete the unverified secondary email if it exists and proceed with account creation.
+            return deleteSecondaryEmailIfUnverified()
           }
+
+          throw err
+        }
+
+        function deleteSecondaryEmailIfUnverified() {
+          return db.getSecondaryEmail(email)
+            .then((secondaryEmailRecord) => {
+              // Currently, users can not create an account from a verified
+              // secondary email address
+              if (secondaryEmailRecord.isVerified) {
+                throw error.verifiedSecondaryEmailAlreadyExists()
```

Comment:
@vladikoff Yea unfortunately this needs to be a custom error. The content-server needs the ability to distinguish because it displays an option to create an account if it doesn't find the email on the account table. If the email is a secondary email, we should not display that option because an account can't be created from it.